compiler/compiler.go

```diff
@@ -1554,7 +1554,11 @@ func (c *compilerContext) maxSliceSize(elementType llvm.Type) uint64 {
 	// Determine the maximum allowed size for a slice. The biggest possible
 	// pointer (starting from 0) would be maxPointerValue*sizeof(elementType) so
 	// divide by the element type to get the real maximum size.
-	maxSize := maxPointerValue / c.targetData.TypeAllocSize(elementType)
+	elementSize := c.targetData.TypeAllocSize(elementType)
+	if elementSize == 0 {
+		elementSize = 1
```

Comment:
Can't you set maxSize=0 for this case to avoid allocations altogether?

Reply:
I think that is not the same thing. Empty struct still requires allocation, since not `nil`.

Reply:
Correct, they require something in the data pointer so that `slice[:0:0]` remains non-nil. It doesn't have to be a true allocation however, it might as wel be a fixed pointer constant (like the one returned by `runtime.alloc(0)`).